homeassistant/components/isy994/helpers.py

```diff
@@ -113,43 +113,53 @@ def _check_for_insteon_type(
             # on ISY 5.x firmware as it uses the superior NodeDefs method
             subnode_id = int(node.address.split(" ")[-1], 16)
 
-            # FanLinc, which has a light module as one of its nodes.
-            if platform == Platform.FAN and subnode_id == SUBNODE_FANLINC_LIGHT:
-                isy_data.nodes[Platform.LIGHT].append(node)
-                return True
-
-            # Thermostats, which has a "Heat" and "Cool" sub-node on address 2 and 3
-            if platform == Platform.CLIMATE and subnode_id in (
-                SUBNODE_CLIMATE_COOL,
-                SUBNODE_CLIMATE_HEAT,
-            ):
-                isy_data.nodes[Platform.BINARY_SENSOR].append(node)
-                return True
-
-            # IOLincs which have a sensor and relay on 2 different nodes
-            if (
-                platform == Platform.BINARY_SENSOR
-                and device_type.startswith(TYPE_CATEGORY_SENSOR_ACTUATORS)
-                and subnode_id == SUBNODE_IOLINC_RELAY
-            ):
-                isy_data.nodes[Platform.SWITCH].append(node)
-                return True
-
-            # Smartenit EZIO2X4
-            if (
-                platform == Platform.SWITCH
-                and device_type.startswith(TYPE_EZIO2X4)
-                and subnode_id in SUBNODE_EZIO2X4_SENSORS
-            ):
-                isy_data.nodes[Platform.BINARY_SENSOR].append(node)
-                return True
-
-            isy_data.nodes[platform].append(node)
-            return True
+            return _check_node_type(isy_data, device_type, platform, subnode_id, node)
```

Comment:
It's a bit weird that _check_node_type always returns true which is returned here on this line. Just letting _check_node_type return void/None and having the call to it followed by return True would have made more sense.